### ISSUE TITLE
Setup GitHub Actions for CD

### ISSUE THREAD

**Issue by amirhoseinsalimi** [amirhoseinsalimi/programmer-fa#58]

**Comment by shywn-mrk** [comment]
I would be happy to contribute.
Can you provide me more information about it?

**Comment by amirhoseinsalimi** [comment]
Wow. I've just opened this. How did you find this repo so quickly?

I just want to setup auto-deployment for this project, so whenever a commit is made to the master branch, it gets deployed on the production server.

I think one of the key points here is going to be tests, so I should write more tests to be more confident.

Now, what is your opinion about this? What should we do before making this happen?

**Comment by shywn-mrk** [comment]
Having tests such as unit tests can be checked before deploying to production. A simple CI/CD for this project can be written in a short amount of time but I need more details.
Writing tests, however, is not related to the pipelines. Our concerns should be separated.

**Comment by amirhoseinsalimi** [comment]
I agree. I don't have much knowledge to create an action script for GitHub actions. Could you write one?

**Comment by amirhoseinsalimi** [comment]
I forgot something. What kind of information do you need? Getting started with the project is already documented in the README. What else do you need?

**Comment by shywn-mrk** [comment]
Where should we exactly deploy? The runners should be configured.

**Comment by amirhoseinsalimi** [comment]
On my personal server. How can we co-operate?

**Comment by shywn-mrk** [comment]
Currently, I'm on vacation but we can have a meeting. I will contact you next week.
